src/core/data/sagas/fetchEntitySetData.js

```diff
@@ -0,0 +1,73 @@
+/*
+ * @flow
+ */
+
+import {
+  call,
+  put,
+  select,
+  takeEvery,
+} from '@redux-saga/core/effects';
+import { Set } from 'immutable';
+import {
+  DataApiActions,
+  DataApiSagas,
+} from 'lattice-sagas';
+import { Logger } from 'lattice-utils';
+import type { Saga } from '@redux-saga/core';
+import type { UUID } from 'lattice';
+import type { WorkerResponse } from 'lattice-sagas';
+import type { SequenceAction } from 'redux-reqseq';
+
+import { FETCH_ENTITY_SET_DATA, fetchEntitySetData } from '../actions';
+import { selectStoredEntityKeyIds } from '../utils';
+
+const LOG = new Logger('DataSagas');
+
+const { getEntitySetData } = DataApiActions;
+const { getEntitySetDataWorker } = DataApiSagas;
+
+function* fetchEntitySetDataWorker(action :SequenceAction) :Saga<*> {
+
+  try {
+    yield put(fetchEntitySetData.request(action.id, action.value));
+
+    const { entityKeyIds, entitySetId } :{
+      entityKeyIds :Set<UUID>;
+      entitySetId :UUID;
+    } = action.value;
+
+    // TODO: expire stored data
+    const storedEntityKeyIds :Set<UUID> = yield select(selectStoredEntityKeyIds(entitySetId, entityKeyIds));
+    const missingEntityKeyIds :Set<UUID> = entityKeyIds.subtract(storedEntityKeyIds);
```

Comment:
No change required, just thinking out loud. The reason I did this originally was to avoid ALWAYS making a request to get data, even if I had JUST made that same request. In retrospect, I think this was a case of premature optimization and we should reconsider this approach.

It would be really great to explore using `react-query` or `rtk-query` for proper caching and such.

Reply:
https://jira.openlattice.com/browse/APPS-2910